src/formats/notion/convert-to-md.ts

```diff
@@ -434,13 +437,62 @@ function splitBrsInFormatting(body: HTMLElement, tagName: FormatTagName) {
 	body.innerHTML = htmlString;
 }
 
-function replaceTableOfContents(body: HTMLElement) {
-	const tocLinks = body.findAll('a[href*=\\#]') as HTMLAnchorElement[];
-	for (const link of tocLinks) {
-		if (link.getAttribute('href')?.startsWith('#')) {
-			link.setAttribute('href', '#' + link.textContent);
-		}
+
+function getTOCIndent(tocItem: Element | null): Number {
+	return Number(tocItem?.classList[1].slice(-1) ?? -1);
+}
+
+function appendTOCItem(itemNew: Element, itemPre: Element | null | undefined) {
+	if (!itemPre) return;
+	const indentNew = getTOCIndent(itemNew);
+	let indentPre = getTOCIndent(itemPre);
+	// itemPre will contain minimum 1 <span> element
+	if (indentNew > indentPre && itemPre.childElementCount == 1) {
+		const listNew = createEl('ul');
+		listNew.append(itemNew);
+		itemPre.append(listNew);
+	}
+	else if (indentNew > indentPre && itemPre.childElementCount == 2) {
+		itemPre.lastElementChild?.append(itemNew);
+	}
+	else if (indentNew == indentPre) {
+		itemPre.parentElement?.append(itemNew);
+	}
+	else if (indentNew < indentPre) {
+		const listPre = itemPre.parentElement;
+		itemPre = listPre?.parentElement;
+		// recurse through parents until (indentNew < indentPre)
```

Comment:
Should this comment be "until (indentNew == indentPre)"?

Reply:
Corrected.